st3/sublime_lib/output_panel.py

```diff
@@ -1,12 +1,70 @@
 from .view_stream import ViewStream
 from .view_utils import set_view_options, validate_view_options
+from ._util.guard import define_guard
 
+__all__ = ['Panel', 'OutputPanel']
 
-__all__ = ['OutputPanel']
 
+class Panel():
+    """An abstraction of a panel, such as the console or an output panel.
 
-class OutputPanel(ViewStream):
-    """A :class:`~sublime_lib.view_stream.ViewStream`
+    :raise ValueError: if `window` has no panel called `panel_name`.
+
+    All :class:`Panel` methods except for :meth:`exists()`
+    will raise a ``ValueError`` if the panel does not exist.
+    Descendant classes may override :meth:`exists()` to customize this behavior.
+    """
+
+    def __init__(self, window, panel_name):
+        self.window = window
+        self.panel_name = panel_name
```

Comment:
This attribute should definitely be documented.

Reply:
Done.